### PR TITLE
[Merged by Bors] - feat: a self-adjoint element of a C*-algebra is upper bounded by the algebra map of its norm

### PR DESCRIPTION
**Pull request by dupuisf**
This PR shows that `a ≤ algebraMap ℝ A ‖a‖` for a self-adjoint element `a : A` of a C*-algebra.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean [hunk 1]
```diff
@@ -282,6 +283,25 @@ lemma spectrum_star_mul_self_nonneg {b : A} : ∀ x ∈ spectrum ℝ (star b * b
   have h_eqOn := eqOn_of_cfc_eq_cfc (a := star b * b) h_eq_a_neg
   simpa [sup_eq_left.mpr hx'.le] using h_eqOn hx
 
+lemma IsSelfAdjoint.le_algebraMap_norm_self [PartialOrder A] [StarOrderedRing A] (a : A)
+    (ha : IsSelfAdjoint a := by cfc_tac) : a ≤ algebraMap ℝ A ‖a‖ := by
```
Comment (j-loreaux) [inline Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean:287]: Also, can we get the non-unital version? That is, `{a : A} (ha : IsSelfAdjoint a) (b : A) : star b * a * b ≤ ‖a‖ • star b * b`? You can prove it by passing to the unitization. (It's possible you need mine an Anatole's work first, in which case we can leave it to a future PR.)

File: Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean [hunk 2]
```diff
@@ -741,6 +741,16 @@ lemma algebraMap_le_cfc (f : R → R) (r : R) (a : A) (h : ∀ x ∈ spectrum R
     algebraMap R A r ≤ cfc f a :=
   cfc_const r a ▸ cfc_mono h
 
+lemma le_algebraMap_of_spectrum_le (r : R) (a : A) (h : ∀ x ∈ spectrum R a, x ≤ r)
```
Comment (j-loreaux) [inline Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean:744]: Unless you think this is a bad idea for some reason
```suggestion
lemma le_algebraMap_of_spectrum_le {r : R} {a : A} (h : ∀ x ∈ spectrum R a, x ≤ r)
```

File: Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean [hunk 3]
```diff
@@ -741,6 +741,16 @@ lemma algebraMap_le_cfc (f : R → R) (r : R) (a : A) (h : ∀ x ∈ spectrum R
...
+    (ha : p a := by cfc_tac) : a ≤ algebraMap R A r := by
+  rw [← cfc_id R a]
+  exact cfc_le_algebraMap id r a h
+
+lemma algebraMap_le_of_le_spectrum (r : R) (a : A) (h : ∀ x ∈ spectrum R a, r ≤ x)
```
Comment (j-loreaux) [inline Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean:749]:
```suggestion
lemma algebraMap_le_of_le_spectrum {r : R} {a : A} (h : ∀ x ∈ spectrum R a, r ≤ x)
```

**Review by j-loreaux — COMMENTED** [review]
I think the non-unital version has to wait for a week or so.

File: Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean [hunk 4]
```diff
@@ -7,6 +7,7 @@ import Mathlib.Analysis.NormedSpace.Star.ContinuousFunctionalCalculus.Restrict
 import Mathlib.Analysis.NormedSpace.Star.ContinuousFunctionalCalculus
 import Mathlib.Analysis.NormedSpace.Star.Spectrum
 import Mathlib.Topology.ContinuousFunction.UniqueCFC
+import Mathlib.Topology.ContinuousFunction.StarOrdered
```
[the remark this answers is not in the document]
Comment (dupuisf) [reply Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean:10]: Yes, fair enough.

File: Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean (hunk 1 above)
Comment (dupuisf) [reply Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean:287]: I'll try it and see how it goes.

Comment (dupuisf) [reply Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean:287]: I'll take your word for it and leave it for another PR.

**Comment by j-loreaux** [comment]
Thanks!

bors merge

**Pull request closed without merging**